proto/src/proto.rs

```diff
@@ -418,6 +419,33 @@ pub struct LdapMatchingRuleAssertion {
     pub dn_attributes: bool, // DEFAULT FALSE
 }
 
+impl LdapMatchingRuleAssertion {
+    pub fn from_strings(left: String, right: String) -> Self {
+        let match_value = right.to_string();
+        let mut split = left.split(':').collect::<VecDeque<_>>();
+        let dn_attribute = split.contains(&"dn");
+        split.retain(|s| *s != "dn");
+        let first = split.pop_front().unwrap();
```

Comment:
Issue: avoid `unwrap` unless you can prove it won’t cause a panic. If you can prove it, add an inline code comment explaining why.

On that note, I think this project is _denying_ `unwrap`s (there is `#[deny(clippy::unwrap_used)]` in lib.rs). I understand clippy is intended to be run during development and all lints fixed. Does that sound accurate? @Firstyear

Reply:
Yep that's correct. If an error can occur we should handle it cleanly, or prove it can never occur.

Reply:
Fair enough, it’s safe in the way it is used today given that we know all the inputs. I see you refactored to avoid the unwrap completely, which is probably safer.

FYI, when handling stuff which could cause a panic like this, I like to establish local invariants in order to clearly convey the intention and why the code is correct (and what should be considered when modifying it).
More in this: https://github.com/Devolutions/IronRDP/blob/0954f42519fd6942f7e04fb692467a4f1b387038/STYLE.md#invariants